### DISCUSSION TITLE
Why is work with customfields so hard ?

### DISCUSSION THREAD

**Discussion by askme23** [andygrunwald/go-jira#622] [Q&A] [answered]
Hey ) Can you explain why does complex custom fields cast with fmt.Sprint ? https://github.com/andygrunwald/go-jira/blob/v1.13.0/issue.go#L1162 
How to work properly with nested value ?!

**Comment by andygrunwald** [comment, marked as answer]
The short answer:
Because we didn't know it better 7 years ago - but today, we know the implementation is not good at all.

The long answer:
As mentioned in the README (https://github.com/andygrunwald/go-jira#warning-state-of-this-library-warning), we are working right now on v2 of this library.
This means, there will be a few breaking changes with the target to fix those behaviors.
Goal is to make it as easy as accessing normal struct fields (for custom fields).

**Reply by askme23** [reply]
Hey, i got you and thx for explanation ) Do you need any help with implementation of this feature ?

**Reply by andygrunwald** [reply]
Happy to accept tickets that discuss a proper design and with some dummy code.
I did not investigate what this can look like, hence have no idea yet.
If you want to write down your thoughts and how a user can easily use this, this would help a lot.
If you want to get some dummy code up and running, also happy to review this.

Not sure if a full implementation from the get-go would lead to a successful merge, but a bit of design discussion might be good. What do you think?

**Reply by askme23** [reply]
Yeah, it is a good point, i try to spend more time this problem later and maybe will write my thoughts in ticket.

**Reply by andygrunwald** [reply]
Awesome. Thank you @askme23 - I will try to respond quickly.
Let me know if anything else is needed.
